Add examples + explanation naming convention and product link rules (#587)
akeneo/pim-helpcenter: content/md/assets/about-product-link-rules.md

The idea of the naming convention feature is to be able to extract those pieces of information and use them to **automatically** enrich your assets with new attributes values.

> Just a reminder but an important one for you to better understand what is coming next: ***the product link rule is based on asset attributes*** :wink:

Reviewer: Je review sur Github et je vois trois '*', normalement on devrait en avoir que deux non?
```suggestion
Just a reminder but an important one for you to better understand what is coming next: **the product link rule is based on asset attributes** :wink:
```

Author: nope, je mets en italique + gras :) donc c'est normal.

> By defining a naming convention, for each [asset family](what-about-assets.html#what-is-an-asset-family), the PIM will be able to split the **asset code** or the **main media filename**, in order to extract the information you want and use it to populate asset attributes.

Reviewer: ```suggestion
Defining a naming convention for each [asset family](what-about-assets.html#what-is-an-asset-family) will enable the PIM to split your **asset code** or your **main media filename** to extract the information you want and use it to populate asset attributes.
```

![Assets Naming Convention](../img/Assets_NamingConventionNew.png)

### Should I split the asset code or the filename?

As said above, you can choose to split the asset code or the main media filename. What should you do? Let's discover it.

#### First use case
If you **mass upload** your assets in the Asset Manager - meaning that your main media attribute is a `media file` attribute - you can choose to **split either the code of the asset, or its filename**.   

#### Second use case
If you create your assets **one by one** in your family, or if the main media attribute of your asset family is a `media link` attribute (meaning that you **can't use the mass upload** feature), you must split the **code** of your assets, **not the filename**.

*Why?*

> Because the naming convention is executed at the ***asset creation*** step. In the second use case, we don't know the filename of the asset at the asset creation since we add the file afterward. That's why the naming convention won't work if you split the filename instead of the code :wink:

Reviewer 2: ```suggestion
Because the naming convention is executed at the ***asset creation*** step. In the second use case, we don't know the filename of the asset when the asset creation takes place since we add the file afterward. That's why the naming convention won't work if you split the filename instead of the code :wink:
```

> Looks difficult? Don't freak out! The following sections are here to help you understand this rule and how you can make the most of it. You'll see, it's super powerful! 😃

Reviewer: I'd rather remove anxiogenic words like 'difficult' and 'freak' ;)
```suggestion
If you are not comfortable with the naming conventions yet, the following sections will help you understand the rules and how you can make the most of it. You'll see, it's super powerful! 😃
```
